content/articles/write-a-rust-graphql-server-with-mongodb-juniper-and-actix-web/index.md

```diff
@@ -0,0 +1,461 @@
+
+Go ahead and create a project directory. Then launch a terminal that points to the created project directory. Finally, run the following command to initialize the project:
+
+```bash
+cargo init
+```
+
+Test the default template project created by running the following command:
+
+```bash
+cargo run
+```
+
+The project will build and then log the `Hello, world!` text. This implies your Rust application is well set, and we can proceed by building our GraphQL server. To set this application stack, we will need to install a couple of dependencies that will help us set up a Rust server and communicate with the MongoDB database. We will use the following dependencies:
+
+- [Actix-Web framework](https://crates.io/crates/actix-web): This is a Rust framework that helps you set up and manage a Rust HTTP server.
+- [Dotenv](https://crates.io/crates/dotenv): To connect to a MongoDB database, we will need to set up environment variables that host the MongoDB connection parameters, such as MongoDB connection URL. Dotenv is used to load the environmental variables to project files.
+- [Features](https://doc.rust-lang.org/cargo/reference/features.html): For handling asynchronous calls to MongoDB.
+- [MongoDB](https://crates.io/crates/mongodb): We are using a MongoDB database. Thus we need a MongoDB driver to handle communication between the GraphQL server and MongoDB database.
+- [serde_json](https://crates.io/crates/serde_json): When sending data to MongoDB, we need to serialize it to JSON format. In this case, we will use serde_json to get the GraphQL API requests and convert the data being sent into JSON format.
+- [Juniper framework](https://docs.rs/juniper/latest/juniper/): This is a GraphQL server framework for Rust programming language. Juniper will help us write a GraphQL server in Rust. It provides type-safe GraphQL APIs and convenient schemas definitions for Rust.
+
+To use the above dependencies, we need to make them available for our project. Navigate to the `Cargo.toml` file in the project root directory and update is as follows:
+
+```rust
+[dependencies]
+actix-web = "1.0.0"
+mongodb = "1.2.0"
+serde_json = "1.0"
+futures = "0.1"
+dotenv = "0.9.0"
+juniper = "0.13.1"
+```
+
+Then install them by running:
+
+```bash
+cargo run
+```
+
+The dependencies will be installed. This will still give you the output `Hello, world!` since we have not added our code logic yet. And this implies you are good to go to the next step.
+
+### Setting up the schema
+Schema is a collection of types with fields that are populated from your back-end application. They are the defining GraphQL API blueprint. Apart from fields, a schema also defines types such as query and mutation. Query and mutation are just requests a client can make to access the API data. A query type sets the API read operation, commonly known as the GET request. A mutations type sets the API write operations, such as POST and PUT requests.
```

Comment:
This statement is not clear. It also incomplete. Correct it, make it clear and break it down into smaller readable paragraphs.